src/coreclr/jit/fgopt.cpp

```diff
@@ -5311,6 +5363,13 @@ bool Compiler::ThreeOptLayout::RunThreeOptPass(BasicBlock* startBlock, BasicBloc
         {
             AddNonFallthroughPreds(startPos + part1Size + part2Size + part3Size);
         }
+
+#ifdef DEBUG
+        // Ensure the swap improved the overall layout
+        const weight_t newLayoutCost = GetLayoutCost(startPos, endPos);
+        assert(newLayoutCost < currLayoutCost);
```

Comment:
Seems like we could end up equal?
```suggestion
        assert(newLayoutCost <= currLayoutCost);
```

Reply:
We'll only perform a swap if the new cost is strictly less than the original cost (plus epsilon), so I don't think it should be possible for the new layout cost to be the same as the old one after swapping, right?